lib/filterx/expr-plus-generator.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2024 Axoflow
+ * Copyright (c) 2024 shifter
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "expr-plus.h"
+#include "object-string.h"
+#include "filterx-eval.h"
+#include "scratch-buffers.h"
+#include "expr-generator.h"
+#include "object-list-interface.h"
+#include "object-dict-interface.h"
+
+typedef struct FilterXOperatorPlusGenerator
+{
+  FilterXExprGenerator super;
+  FilterXBinaryOp elts;
```

Comment:
This seems like a double inheritance, which is not fully supported.

Both `FilterXExprGenerator` and `FilterXBinaryOp` are `FilterXExpr`s, but `FilterXBinaryOp`'s eval is never used. I think it would be cleaner to just store the 2 `FilterXExpr`s and not use `FilterXBinaryOp` at all.